src/main/java/net/glowstone/util/GlowTravelAgent.java

```diff
@@ -0,0 +1,324 @@
+                            // for portal placement
+                            if (distance < 0.0D || newDist < distance) {
+                                distance = newDist;
+                                actualX = tempX;
+                                actualY = tempY;
+                                actualZ = tempZ;
+                                axis = riv % 2;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        int finalX = actualX;
+        int finalY = actualY;
+        int finalZ = actualZ;
+        int coEff1 = axis % 2;
+        int coEff2 = 1 - coEff1;
+
+        if (axis % 4 >= 2) {
+            coEff1 = -coEff1;
+            coEff2 = -coEff2;
+        }
+
+        // Id still no place is found, create an obsidian
+        // platform in the void to place the portal onto
+        if (distance < 0.0D) {
+            actualY = Math.min(Math.max(actualY, 70), worldHeight - 10);
+            finalY = actualY;
+
+            for (int safeBeforeAfter = -1; safeBeforeAfter <= 1; ++safeBeforeAfter) {
+                for (int safeWidth = 0; safeWidth < 2; ++safeWidth) {
+                    for (int height = -1; height < 3; ++height) {
+                        int curX = finalX + safeWidth * coEff1 + safeBeforeAfter * coEff2;
+                        int curY = finalY + height;
+                        int curZ = finalZ + safeWidth * coEff2 - safeBeforeAfter * coEff1;
+                        new Location(world, curX, curY, curZ).getBlock()
+                            .setType(height < 0 ? Material.OBSIDIAN : Material.AIR);
+                    }
+                }
+            }
+        }
+
+        // calculate portal axis based on coefficient 1
+        byte meta = (byte) (coEff1 == 0 ? 2 : 1);
+        ArrayList<Block> blocks = new ArrayList<>(6);
+        for (int width = -1; width < 3; ++width) {
+            for (int height = -1; height < 4; ++height) {
+                int curX = finalX + width * coEff1;
+                int curY = finalY + height;
+                int curZ = finalZ + width * coEff2;
+                Block block = new Location(world, curX, curY, curZ).getBlock();
+                if (width == -1 || width == 2 || height == -1 || height == 3) {
+                    block.setType(Material.OBSIDIAN);
+                } else {
+                    blocks.add(block);
+                }
+            }
+        }
+        PortalCreateEvent event =
+            new PortalCreateEvent(blocks, world, PortalCreateEvent.CreateReason.OBC_DESTINATION);
+        if (!EventFactory.getInstance().callEvent(event).isCancelled()) {
+            event.getBlocks().forEach(block ->
+                block.setTypeIdAndData(Material.PORTAL.getId(), meta, false));
+            return true;
+        }
+        return false;
+    }
+
+    private boolean canBuildPortal(Location loc, BlockFace facing) {
+        // TODO: Need impl
```

Comment:
Can you document what this method is supposed to do?

Reply:
Removed in next commit